sdcm/cluster.py

```diff
@@ -1463,7 +1463,9 @@ def download_scylla_repo(self, scylla_repo):
             self.remoter.run('sudo chmod 644 %s' % repo_path)
             result = self.remoter.run('cat %s' % repo_path, verbose=True)
             verify_scylla_repo_file(result.stdout, is_rhel_like=True)
+            self.is_enterprise = 'enterprise' in result.stdout
```

Comment:
why doing this here, isn't there a different code setting this value ?

Reply:
Yes. But recently it's unstable for some yum error. It might be some real epel repo issue. This PR try to avoid the issue by retry.

```
    def scylla_pkg(self):
        if self.is_enterprise is None:
            if self.is_rhel_like():
                result = self.remoter.run("sudo yum search scylla-enterprise", ignore_status=True)
                self.is_enterprise = True if ('scylla-enterprise.x86_64' in result.stdout or
                                              'No matches found' not in result.stdout) else False
            else:
                result = self.remoter.run("sudo apt-cache search scylla-enterprise", ignore_status=True)
                self.is_enterprise = True if 'scylla-enterprise' in result.stdout else False
```

Reply:
this is too hacky, refactor `sdcm.cluster.BaseNode#scylla_pkg` into two methods:
1. transform `is_enterprise` to property (make sure it is cached, i.e. doesn't run the commands on remote)
    try the "old" logic of searching using the debian/yum repo, if it fails apply logic of checking the repo file  
2.  use `is_enterprise` in `sdcm.cluster.BaseNode#scylla_pkg` to generate package name

Reply:
For reusing the repo content from result.stdout. Actually it will replace original logic.

I had moved it inside is_enterprise() as a plan B only for original logic fail.